mesonbuild/mintro.py

```diff
@@ -125,7 +131,160 @@ def list_target_files(target_name, coredata, builddata):
         out.append(i)
     print(json.dumps(out))
 
-def list_buildoptions(coredata, builddata):
+class BuildoptionsOptionHelper:
+    # mimic an argparse namespace
+    def __init__(self, cross_file):
+        self.cross_file = cross_file
+        self.native_file = None
+        self.cmd_line_options = {}
+
+class BuildoptionsInterperter(astinterpreter.AstInterpreter):
+    # Interpreter to detect the options without a build directory
+    # Most of the code is stolen from interperter.Interpreter
+    def __init__(self, source_root, subdir, backend, cross_file=None, subproject='', subproject_dir='subprojects'):
+        super().__init__(source_root, subdir)
+
+        options = BuildoptionsOptionHelper(cross_file)
+        self.cross_file = cross_file
+        self.environment = environment.Environment(source_root, None, options)
```

Comment:
maybe this should use a environment that is shared by all subprojects?

Reply:
I would not do this, since `Environment` stores `coredata` and I want to avoid conflicts between (sub)projects.

Also, see my last comment.